artifact/artifact.sh

```diff
@@ -1,48 +1,89 @@
 #!/bin/bash -eux
-#
-# This code is meant to be executed within a TravisCI build.
+
+# This code is meant to be executed within a CI build.
 
 publish_to_dockerhub() {
     if [ -n "${DOCKER_USERNAME:-}" ] && [ -n "${DOCKER_PASSWORD:-}"  ]; then
-        DOCKER_REPO="tidepool/${TRAVIS_REPO_SLUG#*/}"
-        echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin
 
-        if [ "${TRAVIS_REPO_SLUG:-}" == "tidepool-org/blip" ]; then
+        # Determine CI provider
+        if [ -n "${TRAVIS:-}" ]; then
+            CI_PROVIDER="travis"
+        elif [ -n "${CIRCLECI:-}" ]; then
+            CI_PROVIDER="circle"
+        else
+            echo "No known CI provider detected"
+            return 1
+        fi
+
+        # Set common variables
+        if [ "$CI_PROVIDER" = "travis" ]; then
+            REPO_SLUG=$TRAVIS_REPO_SLUG
+            COMMIT=$TRAVIS_COMMIT
+            TAG=${TRAVIS_TAG:-}
+            BRANCH=${TRAVIS_BRANCH:-}
+            PR_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-}
+        elif [ "$CI_PROVIDER" = "circle" ]; then
+            REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            COMMIT=$CIRCLE_SHA1
+            TAG=${CIRCLE_TAG:-}
+            BRANCH=${CIRCLE_BRANCH:-}
+            PR_BRANCH=${CIRCLE_PR_NUMBER:-}
+        fi
+
+        if [ -z "$PR_BRANCH" ]; then
+            IS_PULL_REQUEST=false
+        else
+            IS_PULL_REQUEST=true
+        fi
+
+        DOCKER_REPO="tidepool/${REPO_SLUG#*/}"
+
+        echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin
+        if [ "${REPO_SLUG:-}" == "tidepool-org/blip" ] || [ "${REPO_SLUG:-}" == "tidepool-org/uploader" ]; then
+            # Build blip image
             RX_ENABLED=${RX_ENABLED-false}
-            if [[ ",${RX_ENABLED_BRANCHES:-}," = *",${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH},"* ]]; then RX_ENABLED=true; fi
-            DOCKER_BUILDKIT=1 docker build --tag "${DOCKER_REPO}" --build-arg ROLLBAR_POST_SERVER_TOKEN="${ROLLBAR_POST_SERVER_TOKEN:-}" --build-arg LAUNCHDARKLY_CLIENT_TOKEN="${LAUNCHDARKLY_CLIENT_TOKEN:-}" --build-arg REACT_APP_GAID="${REACT_APP_GAID:-}" --build-arg RX_ENABLED="${RX_ENABLED:-}" --build-arg TRAVIS_COMMIT="${TRAVIS_COMMIT:-}" .
+            if [[ ",${RX_ENABLED_BRANCHES:-}," = *",${PR_BRANCH:-$BRANCH},"* ]]; then RX_ENABLED=true; fi
```

Comment:
Is this going to work if we use the PR number for `PR_BRANCH`? I *think* we need to use just `BRANCH` here.

Reply:
I believe you're correct. This check needs to be just against `BRANCH`